### PR TITLE
Reading network from config file.

### PR DESCRIPTION
**Pull request by kaivex**
Host/port is read from the config file. If it is not present there, the system logs a warning and does not start.

While the config file supports multiple hosts, ports, and types of connections, the architecture of the codebase currently does not. It will cheerfully read them, but it only creates a single connection, on the first telnet host/port listed. All others are ignored.

### CODE REVIEW

File: Mural/ListenerConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+			private set;
+		}
+		
+		public String Type
+		{
+			get;
+			private set;
+		}
+			
+		public ListenerConfiguration ()
```
Comment (jwodicka) [inline Mural/ListenerConfiguration.cs]: Move this method to the top of the file, please! I'm not sure what coding standard we're going to settle on, but I think constructors-at-top is a more readable standard than the alternative.

Also, please add some minimal whole-class comments (ideally of the three-slash variety that the IDE can read) explaining why ListenerConfiguration exists.

File: Mural/Main.cs [hunk 2]
```diff
@@ -21,36 +22,74 @@ namespace Mural
 			// TODO: Replace the logic here that connects components with an IoC system.
 			// Probably use Ninject for this purpose: http://ninject.org/
 			
-			// TODO: Figure out how Mural actually becomes aware of the IP addresses it serves,
-			// and the ports it should listen on. Perhaps the standard .NET .config XML files?
+			// This is a list, to be populated with all the host/ports we know about.
+			// Is ArrayList the right type here?
```
Comment (jwodicka) [inline Mural/Main.cs]: Nope, ArrayList is not the right type here.

Add System.Collections.Generic to your class, and use List<ListenerConfiguration> for a strongly-typed list of ListenerConfigurations.

File: Mural/Main.cs [hunk 3]
```diff
@@ -21,36 +22,74 @@ namespace Mural
...
+					}
+				}
+			} catch(ConfigurationException e) {
+				// This will cause a ConfigurationException for a duplicate host or port number.
+				// There may be other error conditions that cause ConfigurationExceptions.
+				// Regardless, if we can't read our configuration, we are sad and confused pandas.
+				// Log as an error, then exit.
+				_log.ErrorFormat(e.Message);
+				Environment.Exit(1);
+			}
```
Comment (jwodicka) [inline Mural/Main.cs]: You are 95% of the way to an encapsulated method to read the list of ListenerConfigurations from the config file, here. Refactor this stuff out into a method that returns connectionList; none of this code has dependency on the stuff around it, so it's a nice clean extraction.

File: Mural/PortElement.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		// ...and a type property.
+		// TODO: Validation. This can only support a limited set of types.
+		// We probably want an enum here.
+		// It can be:
+		//   HTTP
+		//   Telnet
+		//   SSH
+		//   ...Jabber? SMTP? etc.
+		// For now, it's just a string.
+		// If we don't have a type listed, we assume telnet.
```
Comment (jwodicka) [inline Mural/PortElement.cs:30]: Hm. Do we want to assume telnet, or do we want to tell it that this string is mandatory, and the user must provide _something_?

Comment (kaivex) [inline Mural/PortElement.cs:30]: We may, however - I am not aware of a way to actually enforce this in code here.
The IsRequired decorator does not appear to actually make the field required. We /could/ add manual checking later on, but as far as this part of the code goes, the best I seem able to do is give it a default value.
IsRequired may have strange requirements for it to actually work; we can investigate further.

Comment (jwodicka) [inline Mural/PortElement.cs:30]: Understood. We'll consider it suboptimal and check it in like this, then. =)

**Pull request merged**